include/functions.inc.php

```diff
@@ -1158,91 +1158,44 @@ function createHelpLink(int $module, int $category): string
-                if (!isAdmin()) {
-                    redirectTo('/?p=index', 101, __LINE__);
-                }
-                $page = $p;
-                break;
-            case 'bank':
-            case 'bioladen':
-            case 'buero':
-            case 'forschungszentrum':
-            case 'gebaeude':
-            case 'marktplatz_liste':
-            case 'marktplatz_verkaufen':
-            case 'plantage':
-            case 'vertraege_liste':
-            case 'vertraege_neu':
-            case 'mafia':
-            case 'statistik':
-            case 'gruppe':
-            case 'gruppe_einstellungen':
-            case 'gruppe_mitgliederverwaltung':
-            case 'gruppe_diplomatie':
-            case 'gruppe_kasse':
-            case 'gruppe_logbuch':
-            case 'gruppe_krieg_details':
-            case 'rangliste':
-            case 'rangliste_spezial':
-            case 'index':
-            case 'impressum':
-            case 'regeln':
-            case 'einstellungen':
-            case 'nachrichten_lesen':
-            case 'nachrichten_liste':
-            case 'nachrichten_schreiben':
-            case 'notizblock':
-            case 'hilfe':
-            case 'profil':
-            case 'special':
-                $page = $p;
-                break;
-            default:
-                $page = 'index';
-                break;
-        }
+function getCurrentPage() :string
+{
+    $adminSeiten = [
+            'admin', 'admin_benutzer', 'admin_benutzer_bearbeiten', 'admin_benutzer_importieren',
+            'admin_gruppe', 'admin_gruppe_bearbeiten', 'admin_test', 'admin_markt',
+            'admin_vertrag', 'admin_vertrag_einstellen', 'admin_vertrag_bearbeiten',
+            'admin_markt_einstellen', 'admin_markt_bearbeiten', 'admin_log_bank',
+            'admin_log_bioladen', 'admin_log_gruppenkasse', 'admin_log_login',
+            'admin_log_mafia', 'admin_log_marktplatz', 'admin_log_nachrichten',
+            'admin_log_vertraege'
+    ];
+
+    $userSeiten = [
+            'bank', 'bioladen', 'buero', 'forschungszentrum', 'gebaeude', 'marktplatz_liste',
+            'marktplatz_verkaufen', 'plantage', 'vertraege_liste', 'vertraege_neu', 'mafia',
+            'statistik', 'gruppe', 'gruppe_einstellungen', 'gruppe_mitgliederverwaltung',
+            'gruppe_diplomatie', 'gruppe_kasse', 'gruppe_logbuch', 'gruppe_krieg_details',
+            'rangliste', 'rangliste_spezial', 'index', 'impressum', 'regeln', 'einstellungen',
+            'nachrichten_lesen', 'nachrichten_liste', 'nachrichten_schreiben', 'notizblock',
+            'hilfe', 'profil', 'special'
+    ];
+
+    $gastSeiten = ['anmelden', 'registrieren', 'index', 'passwort_vergessen', 'regeln', 'impressum'];
+
+    $aktuelleSeite = array_key_exists('p', $_GET) ? $_GET['p'] : 'index';
+    $istEingeloggt = isLoggedIn();
+    $istAdmin = isAdmin();
+
+    if ($istEingeloggt && $istAdmin && in_array($aktuelleSeite, $adminSeiten)) {
```

Comment:
Wenn man Admin ist, ist man auch eingeloggt ($istEingeloggt redundant)
```suggestion
    if ($istAdmin && in_array($aktuelleSeite, $adminSeiten)) {
```